Mathlib/Order/CompleteBooleanAlgebra.lean

```diff
@@ -415,6 +415,32 @@ theorem iSup_inf_of_antitone {ι : Type*} [Preorder ι] [IsDirected ι (swap (·
     (hf : Antitone f) (hg : Antitone g) : ⨆ i, f i ⊓ g i = (⨆ i, f i) ⊓ ⨆ i, g i :=
   @iSup_inf_of_monotone α _ ιᵒᵈ _ _ f g hf.dual_left hg.dual_left
 
+theorem isGreatest_himp {α} [GeneralizedHeytingAlgebra α] (a b : α) :
+    IsGreatest {w | w ⊓ a ≤ b} (a ⇨ b) := by
+  simp only [IsGreatest, mem_setOf_eq, himp_inf_self, inf_le_left, mem_upperBounds, le_himp_iff,
+    imp_self, implies_true, and_self]
+
+theorem isLeast_sdiff {α} [GeneralizedCoheytingAlgebra α] (a b : α) :
+    IsLeast {w | a ≤ b ⊔ w} (a \ b) := by
+  simp only [IsLeast, mem_setOf_eq, sup_sdiff_self, le_sup_right, mem_lowerBounds, sdiff_le_iff,
+    imp_self, implies_true, and_self]
+
+lemma isGreatest_sSup {α} [CompleteSemilatticeSup α] (a : α) (s : Set α) :
+  IsGreatest s a → sSup s = a :=
+    fun h ↦ IsLUB.sSup_eq (IsGreatest.isLUB h)
+
+lemma isLeast_sInf {α} [CompleteSemilatticeInf α] (a : α) (s : Set α) :
+  IsLeast s a → sInf s = a :=
+    fun h ↦ IsGLB.sInf_eq (IsLeast.isGLB h)
```

Comment:
I doubt we need those lemmas as they are a straightforward combination of two simpler lemmas